interface/main.cpp

```diff
@@ -0,0 +1,55 @@
+#include <fstream>
+#include <iostream>
+#include <cstring>
+
+#include "class_factory.h"
+#include "ilogger.h"
+#include "network_interface.h"
+
+
+void print_usage(const std::string& program_name)
+{
+    std::cerr << "Usage: " << program_name << " interface ip_address keepalived_state" << std::endl
+              << "interface: a required argument" << std::endl << "ip_address: a required argument" << std::endl
+              << "netmask: an optional argument, defaults to '255.255.255.0'" << std::endl
+              << "keepalived_state: must be one of FAULT, BACKUP, MASTER" << std::endl;
+}
+
+int main(int argc, char* argv[])
+{
+    const std::string program_name = argv[0];
+    if (argc < 6)
+    {
+        print_usage(program_name);
+        return 1;
+    }
+
+    const std::string mode = argv[argc - 1];
+    if ((mode != "FAULT") && (mode != "BACKUP") && (mode != "MASTER"))
+    {
```

Comment:
multiple usages of strings fault, backup and master in multiple places directly (here, usage, forward in this file), will be hard to maintain if something changes, i.e. from ALL_CAPITAL to Capital_First notation. This should probably be delegated to a variable